components/eventing-controller/api/v1alpha2/subscription_types.go

```diff
@@ -79,6 +80,14 @@ func (s Subscription) MarshalJSON() ([]byte, error) {
 	return json.Marshal(a)
 }
 
+func (s *Subscription) GetMaxInFlightMessages() int {
+	val, err := strconv.Atoi(s.Spec.Config[MaxInFlightMessages])
+	if err != nil {
+		return defaultMaxInFlightMessages
```

Comment:
If the user has provided incorrect value then throw error instead of using default value. Otherwise, it would not be consistent with the information visible in the Sub CR.

Reply:
yeah you're right! changed